Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -100,13 +106,51 @@ public BlackboardArtifactNode(BlackboardArtifact artifact) {
         this.setIconBaseWithExtension(ExtractedContent.getIconFilePath(artifact.getArtifactTypeID())); //NON-NLS
     }
 
+    @Override
+    @NbBundle.Messages({"BlackboardArtifactNode.getAction.errorTitle=Error getting actions",
+        "BlackboardArtifactNode.getAction.errorMessage=There was a problem getting actions for the selected result."})
+    public Action[] getActions(boolean context) {
+        List<Action> actionsList = new ArrayList<>();
+        actionsList.addAll(Arrays.asList(super.getActions(context)));
+
+        //see if this artifact has a timestamp in any of the supported attributes
+        boolean hasTimeStamp = false;
+
+        for (ArtifactEventType artEventType : ArtifactEventType.getAllArtifactEventTypes()) {
```

Comment:
I am strongly in favor of moving most of this logic into the Timeline package to encapsulate the logic and minimize public API exposure. Also, consider creating a Timeline package for limited export of public methods, if you have not done so already.